src/main.cc

```diff
@@ -178,6 +178,39 @@ signal_cb(ev_loop *loop, struct ev_signal *w, int revents)
 	tarantool_exit(0);
 }
 
+static void
+signal_sigint_cb(ev_loop *loop, struct ev_signal *w, int revents)
+{
+	(void) loop;
+	(void) w;
+	(void) revents;
+
+	/**
+	 * If running in daemon mode, tarantool exits on SIGINT.
+	 * Complain about possibly sudden and unexpected death.
+	 */
+	if (pid_file) {
```

Comment:
Disabling exit on SIGINT unless `pid_file` is set isn't entirely correct. Consider the following example:

```
tarantool <<EOF
box.cfg{}
EOF
```

With this patch I can stop tarantool started like this neither with CTRL-C nor with CTRL-D. You should disable exit on SIGINT only if console is actually running.

Reply:
Fixed.